### PR TITLE
[singlejar] Various portability fixes for MSVC

### PR DESCRIPTION
**Pull request by rongjiecomputer**
- MSVC does not have `errx` functions, so use `diag_errx` etc. instead.
- Fix format when trying to print `size_t`, use `%zu` so that the function will handle 32/64-bit `size_t` according to target system automatically.
- Adding/guarding a few includes for MSVC.
- MSVC does not have `ssize_t`, so replace it with `ptrdiff_t`

#2241 

/cc @laszlocsomor

### CODE REVIEW

File: src/tools/singlejar/output_jar.cc [hunk 1]
```diff
@@ -119,12 +120,12 @@ int OutputJar::Doit(Options *options) {
     // TODO(asmundak):  Consider going back to sendfile() or reflink
     // (BTRFS_IOC_CLONE/XFS_IOC_CLONE) here.  The launcher preamble can
     // be very large for targets with many native deps.
-    ssize_t byte_count = AppendFile(in_fd, 0, statbuf.st_size);
+    ptrdiff_t byte_count = AppendFile(in_fd, 0, statbuf.st_size);
```
Comment (laszlocsomor) [inline src/tools/singlejar/output_jar.cc]: _Comment from internal reviewer:_

AppendFile is still declared as returning ssize_t, how is this working in MSVC?
IMHO a better approach is to define ssize_t when compiling with MSVC.

Comment (rongjiecomputer) [reply src/tools/singlejar/output_jar.cc]: I have a bigger patch locally I haven't upstream to convert all `ssize_t` and `off_t` to C standard type `ptrdiff_t`.

`ssize_t` is a POSIX type that MSVC does not have. MSVC x64 does have `off_t`, but define as `int32_t` instead of `int64_t` due to legacy reason, but that will prevent us from handling large file. `ptrdiff_t` is more consistent.

For 64 bit Unix system, `ssize_t`, `off_t` and `ptrdiff_t` are all typedef of `long long/int64_t`, so this shouldn't break any Unix code.

Comment (laszlocsomor) [reply src/tools/singlejar/output_jar.cc]: _Forwarded._

Comment (rongjiecomputer) [reply src/tools/singlejar/output_jar.cc]: I will do the change in the larger PR. Done.

Comment (laszlocsomor) [reply src/tools/singlejar/output_jar.cc]: _Forwarded._

File: src/tools/singlejar/combiners.cc [hunk 2]
```diff
@@ -15,6 +15,8 @@
 #include "src/tools/singlejar/combiners.h"
 #include "src/tools/singlejar/diag.h"
 
+#include <cctype>
```
Comment (laszlocsomor) [inline src/tools/singlejar/combiners.cc]: _Internal reviewer comment_: Is this change needed?

File: src/tools/singlejar/output_jar.h [hunk 3]
```diff
@@ -18,6 +18,7 @@
 #include <stdio.h>
 
 #include <cinttypes>
+#include <cstddef>
```
Comment (laszlocsomor) [inline src/tools/singlejar/output_jar.h]: _Internal reviewer comment_: Is this change needed?

File: src/tools/singlejar/combiners.cc (hunk 2 above)
Comment (rongjiecomputer) [reply src/tools/singlejar/combiners.cc]: Yes. Needed by `std::isspace` in this file. glibc probably includes `<cctype>` transitively from other standard header files, but not for MSVC.

File: src/tools/singlejar/output_jar.h (hunk 3 above)
Comment (rongjiecomputer) [reply src/tools/singlejar/output_jar.h]: This is a change that I forgot to move to another PR. Removed.

Comment (laszlocsomor) [reply src/tools/singlejar/output_jar.h]: _forwarded_

File: src/tools/singlejar/combiners.cc (hunk 2 above)
Comment (laszlocsomor) [reply src/tools/singlejar/combiners.cc]: _forwarded_
